pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,277 @@ import (
+
+	// DataSecurityCapabilities holds data security configuration that will be followed by the cohort.
+	DataSecurityCapabilities StorageCohortDataSecurityCapabilities `json:"dataSecurityCapabilities,omitempty"`
+
+	// IOConnectivityCapabilities is the I/O capabilities of the cohort
+	IOConnectivityCapabilities StorageCohortIOConnectivityCapabilities `json:"ioConnectivityCapabilities,omitempty"`
+
+	// IOPerformanceCapabilities enlist I/O performance capabilities supported by the cohort
+	IOPerformanceCapabilities StorageCohortIOPerformanceCapabilities `json:"ioPerformanceCapabilities,omitempty"`
+
+	// DataProtectionCapabilities list data protection capabilities in case of failures
+	DataProtectionCapabilities StorageCohortDataProtectionCapabilities `json:"dataProtectionCapabilities,omitempty"`
+}
+
+// StorageCohortDataStorageCapabilities
+type StorageCohortDataStorageCapabilities struct {
+	// Contains the types of access modes supported by the cohort
+	AccessModes []StorageCohortAccessMode `json:"accessModes,omitempty"`
+
+	// ProvisioningPolicies defines what type of pool/volume can be created inside the cohort.
+	ProvisioningPolicies []ProvisioningPolicy `json:"provisioning_policies,omitempty"`
+
+	// Multipathing describes the different configurations of accessing
+	// storage through different nodes within a cohort
+	Multipathing StorageCohortMultipathingOption `json:"multipathing,omitempty"`
+
+	// If specified, the compression algorithm supported by the cohort
+	// +nullable
+	Compression *[]StorageCohortCompressionAlgorithm `json:"compression,omitempty"`
+
+	// Deduplication, if set to true, ensures that the data is de-duplicated within the cohort
+	// +nullable
+	Deduplication *bool `json:"deduplication,omitempty"`
+}
+
+// StorageCohortDataSecurityCapabilities
+type StorageCohortDataSecurityCapabilities struct {
+	// MediaEncryption either specifies whether encryption is supported or specifies the encryption algorithm itself
+	// +nullable
+	MediaEncryption *[]StorageCohortEncryptionAlgorithm `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy specifies the supported data sanity checks
+	DataSanitizationPolicy StorageCohortDataSanitizationPolicy `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// StorageCohortIOConnectivityCapabilities
+type StorageCohortIOConnectivityCapabilities struct {
+	// Contains the list of access protocols supported by the cohort
+	AccessProtocols []StorageCohortAccessProtocol `json:"accessProtocols,omitempty"`
+}
+
+// StorageCohortIOPerformanceCapabilities
+type StorageCohortIOPerformanceCapabilities struct {
+	// Average I/O operation latency(in microseconds) allowed within the cohort
+	AverageIOOperationLatencyMicroseconds uint64 `json:"averageIoOperationLatencyMicroseconds,omitempty"`
+
+	// Max I/O operation(per second/TB) allowed within the cohort
+	MaxIOOperationsPerSecondPerTerabyte uint64 `json:"maxIoOperationsPerSecondPerTerabyte,omitempty"`
```

Comment:
We need to check if we need it. As it will be part of storage tire.

Reply:
So, should I remove it for now?

Reply:
Lets have these as they are alternate ways of specifying performance capabilities.